OpenRA.Mods.Common/Traits/World/EditorResourceLayer.cs

```diff
@@ -163,8 +163,8 @@ public virtual CellContents UpdateDirtyTile(CPos c)
 			NetWorth += t.Density * type.Info.ValuePerUnit;
 
 			var sprites = type.Variants[t.Variant];
-			var frame = int2.Lerp(0, sprites.Length - 1, t.Density - 1, type.Info.MaxDensity);
-			t.Sprite = sprites[frame];
+			var frame = int2.Lerp(0, sprites.Length, t.Density, type.Info.MaxDensity);
+			t.Sprite = sprites[frame - 1];
```

Comment:
I think it can't crash anymore, because the Density = 0 bug has already been fixed in https://github.com/OpenRA/OpenRA/pull/5571.